File: src/cn/timelives/java/math/MathCalculatorHolder.java
/**
 * 
 */
package cn.timelives.java.math;

/**
 * 
 * Describes a holder of {@link MathCalculator}, which uses the MathCalculator to do operations.
 * @author liyicheng
 *
 */
public interface MathCalculatorHolder<T> extends CalculatorHolder<T, MathCalculator<T>> {
	
	/**
	 * Return the calculator this object is using.
	 * @return a calculator
	 */
	public MathCalculator<T> getMathCalculator();
	
}

File: src/cn/ancono/math/MathCalculatorHolder.kt
/**
 *
 */
package cn.ancono.math

import cn.ancono.math.numberModels.api.RealCalculator

/**
 *
 * Describes a holder of [RealCalculator], which uses the MathCalculator to do operations.
 * @author liyicheng
 */
interface MathCalculatorHolder<T> : CalculatorHolder<T, RealCalculator<T>> {

    override val calculator: RealCalculator<T>

}
